packages/desktop-client/src/components/modals/CreateAccount.tsx

```diff
@@ -16,23 +19,77 @@ import { View } from '../common/View';
 type CreateAccountProps = {
   modalProps: CommonModalProps;
   syncServerStatus: SyncServerStatus;
+  upgradingAccountId?: string;
 };
 
 export function CreateAccount({
   modalProps,
   syncServerStatus,
+  upgradingAccountId,
 }: CreateAccountProps) {
   const actions = useActions();
   const [isGoCardlessSetupComplete, setIsGoCardlessSetupComplete] =
     useState(null);
+  const [isSimpleFinSetupComplete, setIsSimpleFinSetupComplete] =
+    useState(null);
 
-  const onConnect = () => {
+  const onConnectGoCardless = () => {
     if (!isGoCardlessSetupComplete) {
       onGoCardlessInit();
       return;
     }
 
-    authorizeBank(actions.pushModal);
+    if (upgradingAccountId === undefined) {
```

Comment:
nit: Can we do `upgradingAccount == null` here to handle both `null` and `undefined`?